File: src/Tree/BinaryTreePostorderTraversal/BinaryTreePostorderTraversal_Recursive.java
class Solution {
    List<Integer> res = new LinkedList<>();
    
    public List<Integer> postorderTraversal(TreeNode root) {
        dfs(root);
        return res;
    }
    private void dfs(TreeNode root){
        if(root == null) return;
        
        //Traverse the left
        dfs(root.left);
        //Traverse the right
        dfs(root.right);
        //Take the current root
        res.add(root.val);
    }
}

File: src/Tree/BinaryTreePostorderTraversal/BinaryTreePostorderTraversal_Recursive.py
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        
        def dfs(node):
            if node is None:
                return
            
            dfs(node.left)
            
            dfs(node.right)
            
            res.append(node.val)
            
        dfs(root)
        
        return res
